tools/azure-sdk-tools/devtools_testutils/aio/proxy_testcase_async.py

```diff
@@ -69,14 +69,15 @@ async def combined_call(*args, **kwargs):
         test_output = None
         try:
             try:
-                test_output = await test_func(*args, variables=variables, **trimmed_kwargs)
+                # tests don't record successfully unless test_output is a dictionary
+                test_output = await test_func(*args, variables=variables, **trimmed_kwargs) or {}
```

Comment:
This is fine with me if it works, but I would be more curious to see where is the logic that is skipped when `test_output` is `None`? It seems to me that would be the more appropriate place for a fix.

Reply:
That's a good point. We actually send `test_output` to the proxy in the [stop_record_or_playback](https://github.com/Azure/azure-sdk-for-python/blob/main/tools/azure-sdk-tools/devtools_testutils/proxy_testcase.py#L130) method, which would be a more appropriate place to fix this. I'll make that change in a moment

Reply:
In that case, this is a good defense against wonkiness in the test-proxy